populate/main.go

```diff
@@ -0,0 +1,244 @@
+	positionDirection := dextypes.PositionDirection_LONG
+	priceScale := math.Pow(5*math.Pow(rand.Float64(), 3)+1, 2)/100 + 1
+	newPrice := sdk.NewDec(int64(float64(initialLongPrice.RoundInt64()) * priceScale))
+	newQuantity := longQuantity
+
+	if rand.Float64() > 0.5 {
+		positionDirection = dextypes.PositionDirection_SHORT
+		priceScale = 1 - (priceScale - 1)
+		newPrice = sdk.NewDec(int64(float64(initialShortPrice.RoundInt64()) * priceScale))
+		newQuantity = shortQuantity
+	}
+
+	return dextypes.Order{
+		Account:           account,
+		ContractAddr:      contractAddress,
+		PositionDirection: positionDirection,
+		Price:             newPrice,
+		Quantity:          newQuantity,
+		PriceDenom:        "SEI",
+		AssetDenom:        "ATOM",
+		OrderType:         dextypes.OrderType_LIMIT,
+		Data:              VortexData,
+	}
+}
+
+func run(
+	contractAddress string,
+	numberOfAccounts uint64,
+	numberOfBlocks uint64,
+	longPriceFloor uint64,
+	longPriceCeiling uint64,
+	shortPriceFloor uint64,
+	shortPriceCeiling uint64,
+	quantityFloor uint64,
+	quantityCeiling uint64,
+) {
+	grpcConn, _ := grpc.Dial(
+		"127.0.0.1:9090",
+		grpc.WithInsecure(),
+	)
+	defer grpcConn.Close()
+	TxClient = typestx.NewServiceClient(grpcConn)
+	userHomeDir, _ := os.UserHomeDir()
+	filename := filepath.Join(userHomeDir, "outputs", "test_tx_hash")
+	_ = os.Remove(filename)
+	file, err := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		fmt.Printf("Error opening file %s", err)
+		return
+	}
+	TxHashFile = file
+	var mu sync.Mutex
+
+	activeAccounts := []int{}
+	inactiveAccounts := []int{}
+	for i := 0; i < int(numberOfAccounts); i++ {
+		if i%2 == 0 {
+			activeAccounts = append(activeAccounts, i)
+		} else {
+			inactiveAccounts = append(inactiveAccounts, i)
+		}
+	}
+	wgs := []*sync.WaitGroup{}
+	sendersList := [][]func(){}
+	for i := 0; i < int(numberOfBlocks); i++ {
+		fmt.Printf("Preparing %d-th block\n", i)
+		wg := &sync.WaitGroup{}
+		var senders []func()
+		wgs = append(wgs, wg)
+		for j, account := range activeAccounts {
+			key := GetKey(uint64(account))
+			orderPlacements := []*dextypes.Order{}
+			longPrice := uint64(j)%(longPriceCeiling-longPriceFloor) + longPriceFloor
+			longQuantity := uint64(rand.Intn(int(quantityCeiling)-int(quantityFloor))) + quantityFloor
+			shortPrice := uint64(j)%(shortPriceCeiling-shortPriceFloor) + shortPriceFloor
+			shortQuantity := uint64(rand.Intn(int(quantityCeiling)-int(quantityFloor))) + quantityFloor
+			for j := 0; j < BatchSize; j++ {
+				randomOrder := generateRandomOrder(
+					sdk.AccAddress(key.PubKey().Address()).String(),
+					contractAddress, sdk.NewDec(int64(longPrice)).Quo(FromMili),
+					sdk.NewDec(int64(shortPrice)).Quo(FromMili),
+					sdk.NewDec(int64(longQuantity)).Quo(FromMili),
+					sdk.NewDec(int64(shortQuantity)).Quo(FromMili),
+				)
+				orderPlacements = append(orderPlacements, &randomOrder)
```

Comment:
Instead of adding both a long and short order, uses generate `generateRandomOrder` to add just one order